python/paddle/tensor/array.py

```diff
@@ -0,0 +1,150 @@
+
+    Case 1:
+
+    .. code-block:: text
+
+        Input:
+            The shape of first three tensors are [1], and that of the last one is [1,2]:
+                array = ([0.6], [0.1], [0.3], [0.4, 0.2])
+            And:
+                i = [3]
+
+        Output:
+            output = [0.4, 0.2]
+
+    Args:
+        array (list|Tensor): The input array. In dynamic mode, ``array`` is a Python list. But in static mode, array is a Tensor whose ``VarType`` is ``LOD_TENSOR_ARRAY``.
+        i (Tensor): 1-D Tensor, whose shape is [1] and dtype is int64. It represents the
+            specified read position of ``array``.
+
+    Returns:
+        Tensor: A Tensor that is read at the specified position of ``array``.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            arr = paddle.tensor.create_array(dtype="float32")
+            x = paddle.full(shape=[1, 3], fill_value=5, dtype="float32")
+            i = paddle.zeros(shape=[1], dtype="int32")
+
+            arr = paddle.tensor.array_write(x, i, array=arr)
+
+            item = paddle.tensor.array_read(arr, i)
+            print(item)     # [[5., 5., 5.]]
+    """
+    return layers.array_read(array, i)
+
+
+def array_write(x, i, array=None):
+    """
+    This OP writes the input ``x`` into the i-th position of the ``array`` returns the modified array.
+    If ``array`` is none, a new array will be created and returned.
+
+    Args:
+        x (Tensor): The input data to be written into array. It's multi-dimensional
+            Tensor or LoDTensor. Data type: float32, float64, int32, int64.
+        i (Tensor): 1-D Tensor with shape [1], which represents the position into which
+            ``x`` is written.
+        array (list|Tensor, optional): The array into which ``x`` is written. The default value is None,
+            when a new array will be created and returned as a result. In dynamic mode, ``array`` is a Python list.
+            But in static mode, array is a Tensor whose ``VarType`` is ``LOD_TENSOR_ARRAY``.
+
+    Returns:
+        list|Tensor: The input ``array`` after ``x`` is written into.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            arr = paddle.tensor.create_array(dtype="float32")
+            x = paddle.full(shape=[1, 3], fill_value=5, dtype="float32")
+            i = paddle.zeros(shape=[1], dtype="int32")
+
+            arr = paddle.tensor.array_write(x, i, array=arr)
+
+            item = paddle.tensor.array_read(arr, i)
+            print(item)     # [[5., 5., 5.]]
+    """
+    return layers.array_write(x, i, array)
+
+
+def create_array(dtype):
+    """
+    This OP creates an array. It is used as the input of :ref:`api_paddle_tensor_array_read` and
+    :ref:`api_paddle_tensor_array_write`.
+
+    Args:
+        dtype (str): The data type of the elements in the array. Support data type: float32, float64, int32, int64.
```

Comment:
Same as above comment: does it support bool?

Reply:
Yes. `bool` has been added.